project_stock/models/project_project.py

```diff
@@ -34,13 +34,15 @@ class ProjectProject(models.Model):
 
     @api.onchange("picking_type_id")
     def _onchange_picking_type_id(self):
-        self.location_id = self.picking_type_id.default_location_src_id.id
-        self.location_dest_id = self.picking_type_id.default_location_dest_id.id
+        self.location_id = self.picking_type_id.default_location_src_id
+        self.location_dest_id = self.picking_type_id.default_location_dest_id
 
     def write(self, vals):
         """Update location information on pending moves when changed."""
         res = super().write(vals)
         field_names = ("location_id", "location_dest_id")
         if any(vals.get(field) for field in field_names):
-            self.task_ids._update_moves_info()
+            self.task_ids.filtered(
```

Comment:
Why limiting this? Isn't there the possibility of having a desynchronization in any moment due to this?

Reply:
Sorry, no need, I was testing a use case that generated confusion and auto-changed the stock move state causing a side effect not taken into account.